ui/pages/send/send-content/send-hex-data-row/send-hex-data-row.js

```diff
@@ -0,0 +1,57 @@
+import React, { useState } from 'react';
+import { isHexString } from 'ethereumjs-util';
+import { useDispatch, useSelector } from 'react-redux';
+import { getSendHexData, updateSendHexData } from '../../../../ducks/send';
+import SendRowWrapper from '../send-row-wrapper';
+import Dialog from '../../../../components/ui/dialog';
+import { useI18nContext } from '../../../../hooks/useI18nContext';
+
+const SendHexDataRow = () => {
+  const [shouldShowError, setShouldShowError] = useState(false);
+
+  const t = useI18nContext();
+
+  const data = useSelector(getSendHexData);
+
+  const dispatch = useDispatch();
+
+  const onInput = (event) => {
+    const isValidHex = isHexString(event.target.value) || !event.target.value;
+    setShouldShowError(!isValidHex);
+    const hexData = event.target.value.replace(/\n/gu, '') || null;
+    dispatch(updateSendHexData(hexData));
+  };
```

Comment:
This is doing the  same work as the send slice. I'd just set the input in send slice, let the validation happen there, add a new selector to get the hexData.error key and use that to display the error.

Reply:
Fixed [here](https://github.com/MetaMask/metamask-extension/pull/14724/commits/d9627c0fe0550e779669dece355c5c33baa40303)